src/ansys_sphinx_theme/__init__.py

```diff
@@ -107,6 +109,113 @@ def setup_default_html_theme_options(app):
     app.config.html_theme_options.setdefault("collapse_navigation", True)
 
 
+def pv_html_page_context(app, pagename: str, templatename: str, context, doctree) -> None:
+    """Add a function that Jinja can access for returning an "edit this page" link .
+
+    This function will create an "edit this page" link for any library.
+    The link will point to the corresponding file on the main branch.
+
+    Parameters
+    ----------
+    app : Sphinx
+        The Sphinx application instance for rendering the documentation.
+
+    pagename : str
+        The name of the current page.
+
+    templatename : str
+        The name of the template being used.
+
+    context : dict
+        The context dictionary for the page.
+
+    doctree : document
+        The document tree for the page.
+
+    References
+    ----------
+    .. [1] GitHub pull request: https://github.com/pyvista/pyvista/pull/4113
+       (Author: akaszynski <https://github.com/akaszynski>)
+    """
+
+    def fix_edit_link_button(link: str) -> str:
+        """Transform "edit on GitHub" links to the correct URL.
+
+        This function will create the correct URL for the "edit this page" link.
+
+        Parameters
+        ----------
+        link : str
+            The link to the GitHub edit interface.
+
+        Returns
+        -------
+        str
+            The link to the corresponding file on the main branch.
+        """
+        github_user = context.get("github_user", "")
+        github_repo = context.get("github_repo", "")
+        version = getattr(app.builder.env.config, "version", None)
```

Comment:
We should raise a warning in case these two are not available in the context.

Reply:
it is already raising warning by pydata, edit this button will only work if html context is set.